blueflood-core/src/integration-test/java/com/rackspacecloud/blueflood/io/EnumIOIntegrationTest.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2016 Rackspace.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io;
+
+import com.datastax.driver.core.ResultSetFuture;
+import com.google.common.collect.Table;
+import com.rackspacecloud.blueflood.io.astyanax.AstyanaxEnumIO;
+import com.rackspacecloud.blueflood.io.astyanax.AstyanaxReader;
+import com.rackspacecloud.blueflood.io.astyanax.AstyanaxWriter;
+import com.rackspacecloud.blueflood.io.datastax.DatastaxEnumIO;
+import com.rackspacecloud.blueflood.outputs.formats.MetricData;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.types.*;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+/**
+ * Integration tests covering EnumIO
+ */
+public class EnumIOIntegrationTest extends IntegrationTestBase  {
+
+    protected Map<Locator, List<IMetric>> locatorToMetrics;
+    protected DatastaxEnumIO datastaxEnumIO = new DatastaxEnumIO();
+    protected AstyanaxEnumIO astyanaxEnumIO = new AstyanaxEnumIO();
+
+    /**
+     * This method is to supply the granularity parameter to some test methods below
+     *
+     * @return
+     */
+    protected Object getGranularitiesToTest() {
+        return new Object[] {
+                Granularity.FULL,
+                Granularity.MIN_5,
+                Granularity.MIN_20,
+                Granularity.MIN_60,
+                Granularity.MIN_240,
+                Granularity.MIN_1440
+        };
+    }
+
+
+    @Before
+    public void generateEnum() throws Exception {
+        locatorToMetrics = generateEnumForTenants();
+    }
+
+    @RunWith(JUnitParamsRunner.class)
+    public static class WriteDatastaxReadAstyanaxReader extends EnumIOIntegrationTest {
```

Comment:
I am doing this to show I'm cool 😉.

JK, I'm doing it because I can organize the tests into groups this way. They are all testing Enum stuff, so in a way I think they need to be together, they share some common utility code. But the static inner classes groups several tests together. This class groups the tests related to writing with datastax, reading with the old AstyanaxReader. There's another one for writing with datastax, reading with the new AstayanxEnumIO, etc.